extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/pages/databaseSettingsPage.ts

```diff
@@ -128,9 +149,33 @@ export class DatabaseSettingsPage extends BasePage {
 		this._endIpAddressTextRow = this.wizard.createFormRowComponent(view, constants.EndIpAddressLabel, '', this._endIpAddressTextbox, true);
 	}
 
-	private createFirewallNameText(view: azdata.ModelView) {
+	private validateFirewallNameText(firewallname: string | undefined): boolean {
```

Comment:
Where are these rules from? In cases like this it's good to say why you're using these specific rules - with links to documents if they exist.

Reply:
The link you added just brings me to the create rule page with no indication of where on that page you're getting the rules from, which isn't really useful for the context here.

Reply:
And again - please add comments for each check about what it's checking. Regex is not super easy to parse for humans and so having an english description of what it's supposed to be validating helps ensure it's done correctly

Reply:
The link in the doc still isn't very useful. Is there no public documentation of these rules?